torch/_dynamo/utils.py

```diff
@@ -2156,7 +2156,7 @@ def tensor_static_reason_to_message(reason: TensorStaticReason):
 def tensor_always_has_static_shape(
     tensor: Union[torch.Tensor, Any],
     is_tensor: bool,
-    guard_source: "torch._guards.GuardSource",
+    tensor_source: Any,
```

Comment:
Why did this annotation get worse?

Reply:
Just change in the argument. It was a guard_source earlier (which is a specific thing and of type GuardSource). Now it's just source.

Reply:
well we've got a perfectly good Source type we can put here :)